pkg/util/strategicpatch/patch.go

```diff
@@ -31,7 +31,352 @@ import (
 // lists should be merged or replaced.
 //
 // For more information, see the PATCH section of docs/api-conventions.md.
+
+// CreateStrategicMergePatch creates a patch that can be passed to StrategicMergePatch.
```

Comment:
@ghodss, actually, there is a situation where a conflict can arise that cannot be resolved by overriding the changes made by the automated process with the user's changes. It occurs when the automated process has deleted or modified a value that the user has not changed. Applying the patch computed by comparing the original and modified configurations from the user to live won't restore the value to its original state, which is what the user expects. The reason is that the patch won't contain the value deleted or modified by the automated process.

I'm inclined to think that this is the only situation where a conflict should matter, since it's the only situation where we can't produce what the user requested. What do you think?